src/emulator/remoteconfig/cloudFunctions.ts

```diff
@@ -0,0 +1,123 @@
+import * as uuid from "uuid";
+import { EmulatorLogger } from "../emulatorLogger";
+import { Emulators } from "../types";
+import { Client } from "../../apiv2";
+import { EmulatorRegistry } from "../registry";
+import { toSerializedDate } from "../storage/metadata";
+import { CloudEvent } from "../events/types";
+import { RemoteConfigEventData } from "@google/events/firebase/remoteconfig/v1/RemoteConfigEventData";
+
+type RemoteConfigFunctionAction = "update";
+const REMOTE_CONFIG_V2_ACTION_MAP: Record<RemoteConfigFunctionAction, string> = {
+  update: "updated",
+};
+
+export interface RemoteConfigEventPayload {
+  description: string;
+  updateOrigin: string;
+  updateTime: string;
+  updateType: string;
+  updateUser: any;
+  versionNumber: number;
+}
+
+export class RemoteConfigCloudFunctions {
+  private logger = EmulatorLogger.forEmulator(Emulators.REMOTE_CONFIG);
+  private multicastPath = "";
+  private enabled = false;
+  private client?: Client;
+
+  constructor(private projectId: string) {
+    if (EmulatorRegistry.isRunning(Emulators.FUNCTIONS)) {
+      this.enabled = true;
+      this.multicastPath = `/functions/projects/${projectId}/trigger_multicast`;
+      this.client = EmulatorRegistry.client(Emulators.FUNCTIONS);
+    }
+  }
+
+  public async dispatch(
+    action: RemoteConfigFunctionAction,
+    object: RemoteConfigEventPayload
+  ): Promise<void> {
+    if (!this.enabled) {
+      return;
+    }
+
+    const errStatus: Array<number> = [];
+    let err: Error | undefined;
+    try {
+      /** Legacy Google Events */
+      const eventBody = this.createLegacyEventRequestBody(action, object);
+      const eventRes = await this.client!.post(this.multicastPath, eventBody);
```

Comment:
Rather than using a nonnull assertion, consider just adding an if statement to check at the top of the function:
```
if (!this.client) {
  // throw or log and error as you see appropriate
   this.logger.LogLabelledError("ERROR", "remote config", "Tried to dispatch events but did not have a functions client. This is likely a bug with initializing the remote config emulator - please report")
   return;
}
```

Reply:
done